primitives/transaction-pool/src/runtime_api.rs

```diff
@@ -22,22 +22,32 @@ use sp_runtime::traits::Block as BlockT;
 
 sp_api::decl_runtime_apis! {
 	/// The `TaggedTransactionQueue` api trait for interfering with the transaction queue.
-	#[api_version(2)]
+	#[api_version(3)]
 	pub trait TaggedTransactionQueue {
 		/// Validate the transaction.
 		#[changed_in(2)]
 		fn validate_transaction(tx: <Block as BlockT>::Extrinsic) -> TransactionValidity;
 
+		/// Validate the transaction.
+		#[changed_in(3)]
+		fn validate_transaction(
+			source: TransactionSource,
+			tx: <Block as BlockT>::Extrinsic,
+		) -> TransactionValidity;
+
 		/// Validate the transaction.
 		///
 		/// This method is invoked by the transaction pool to learn details about given transaction.
 		/// The implementation should make sure to verify the correctness of the transaction
-		/// against current state.
+		/// against current state. The given `block_hash` corresponds to the hash of the block
+		/// that is used as current state.
```

Comment:
Shall we add info about the need to manually initialize the `frame_system` pallet?

Reply:
For frame this is done automatically through `Executive::validate_transaction`. While this is also just required when you have a signed extension that checks for block hashes.